.github/workflows/uv-lock.yml

```diff
@@ -0,0 +1,24 @@
+# pre-commit.ci does not allow network access, so we run this on github actions
+name: Update uv.lock
+
+on:
+  pull_request_target:
```

Comment:
The `pull_request_target` event runs before GitHub detecting merge conflicts (it runs regardless of whether there are merge conflicts or not), so it’s necessary to check for merge conflicts. This PR should only run if there are no conflicts.